### PR TITLE
feat: add handler to upload file assets to course

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR adds support for uploading, listing and removing images from the course's assets. We're not keeping track of what assets are uploaded with these components as of today, so we list all assets from the course.

### How to test
1. Install this Xblock using this PR (https://github.com/eduNEXT/xblock-imagesgallery/pull/7) rebased with these latest changes
2. Then, try uploading a file using the library
3. Try listing the images from the LMS/CMS Preview
4. Try removing files from the CMS preview

### CODE REVIEW

File: imagesgallery/static/html/imagesgallery.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="imagesgallery_block">
```
Comment (mariajgrimaldi) [inline imagesgallery/static/html/imagesgallery.html]: Note to author: we don't need these changes anymore

File: imagesgallery/imagesgallery.py [hunk 2]
```diff
@@ -55,20 +87,113 @@ def student_view(self, context=None):
...
+        assets, total_count = self._get_assets_for_page(self.course_id, query_options)
+        serialized_assets = []
+        for asset in assets:
+            serialized_assets.append(self.get_asset_json_from_dict(asset))
+        return {
+            "files": serialized_assets,
+            "total_count": total_count,
+        }
+
+    def _get_assets_for_page(self, course_key, options):
```
Comment (mariajgrimaldi) [inline imagesgallery/imagesgallery.py:192]: Note to author: remove _ definitions

File: imagesgallery/static/js/src/imagesgallery.js [hunk 3]
```diff
@@ -1,28 +1,41 @@
 /* Javascript for ImagesGalleryXBlock. */
 function ImagesGalleryXBlock(runtime, element) {
 
-    function updateCount(result) {
```
Comment (mariajgrimaldi) [inline imagesgallery/static/js/src/imagesgallery.js:4]: Note to author: we don't need these changes anymore

**Comment by BryanttV** [comment]
When there aren't any images in the course, the component in the LMS shows this:
image
Maybe we could display some message to the user informing that no image has been added yet.

**Comment by BryanttV** [comment]
Please, correct quality errors.

File: imagesgallery/imagesgallery.py [hunk 4]
```diff
@@ -55,20 +94,113 @@ def student_view(self, context=None):
...
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
+    @XBlock.handler
+    def file_upload(self, request, suffix=''):
+        """Handler for file upload to the course assets."""
+        try:
+            from cms.djangoapps.contentstore.views.assets import update_course_run_asset  # pylint: import-outside-toplevel
+        except ImportError:
+            from cms.djangoapps.contentstore.asset_storage_handler import update_course_run_asset  # pylint: import-outside-toplevel
```
Comment (Ian2012) [inline imagesgallery/imagesgallery.py]: I imagine there is a reason behind this try catch for this import, can you leave a comment explaining it?

**Review by Ian2012 — APPROVED** [review]
Just one nit. LGTM. Can you remove unnecessary code (html, js)?

Can you implement a couple of tests for this class?

**Comment by mariajgrimaldi** [comment]
@Ian2012: I'll add tests once we have the definite implementation (underway this sprint) that doesn't try pulling images from the courses' assets but instead gets the images from the xblock state.

File: imagesgallery/imagesgallery.py (hunk 4 above)
Comment (mariajgrimaldi) [reply imagesgallery/imagesgallery.py]: Of course! Thanks

**Comment by mariajgrimaldi** [comment]
@BryanttV: can you leave the comment in the [react PR](https://github.com/eduNEXT/xblock-imagesgallery/pull/7)? thanks!

**Pull request merged**